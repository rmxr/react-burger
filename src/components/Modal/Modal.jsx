import React from "react";
import ReactDOM from "react-dom";
import styles from "./Modal.module.css"
import {CloseIcon} from "@ya.praktikum/react-developer-burger-ui-components";
import ModalOverlay from "../ModalOverlay/ModalOverlay";
import PropTypes from "prop-types";

const modalRoot = document.getElementById("modal");

function Modal({onClose, children}) {
    React.useEffect(() => {
        const handleEscape = (e) => {
            e.key === "Escape" && onClose();
        };
        document.addEventListener("keydown", handleEscape);
        return () => document.removeEventListener("keydown", handleEscape)
    }, []);
    return ReactDOM.createPortal((
        <>
            <ModalOverlay onClose={onClose} />
            <div className={styles.popup}>
                <div className={styles.closeButton}>
                    <CloseIcon onClick={onClose} type="primary"/>
                </div>
                {children}
            </div>
        </>
    ), modalRoot)
};

Modal.propTypes = {
    onClose: PropTypes.func,
    children: PropTypes.element,
}

export default Modal;