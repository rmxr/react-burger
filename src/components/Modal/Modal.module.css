.popup {
    position: fixed;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    width: 720px;
    min-height: 500px;
    border-radius: 40px;
    border: 1px solid rgba(76, 76, 255, 0.2);
    background-color: rgba(28, 28, 33, 1);
    box-sizing: border-box;
}

.closeButton {
    position: absolute;
    top: 60px;
    right: 40px;
}

.closeButton:hover {
    opacity: .7;
}