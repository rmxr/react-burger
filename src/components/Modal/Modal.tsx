import React from "react";
import ReactDOM from "react-dom";
import styles from "./Modal.module.css"
import {CloseIcon} from "@ya.praktikum/react-developer-burger-ui-components";
import ModalOverlay from "../ModalOverlay/ModalOverlay";

const modalRoot = document.getElementById("modal") as HTMLElement;

function Modal({onClose, children}: { onClose: () => void; children: JSX.Element }) {
  React.useEffect(() => {
    const handleEscape = (e: KeyboardEvent) => {
      e.key === "Escape" && onClose();
    };
    document.addEventListener("keydown", handleEscape);
    return () => document.removeEventListener("keydown", handleEscape)
  }, []);
  return ReactDOM.createPortal((
    <>
      <ModalOverlay onClose={onClose}/>
      <div className={styles.popup}>
        <div className={styles.closeButton}>
          <CloseIcon onClick={onClose} type="primary"/>
        </div>
        {children}
      </div>
    </>
  ), modalRoot)
}

export default Modal;