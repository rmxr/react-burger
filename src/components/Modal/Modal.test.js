import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from '@testing-library/react';
import ReactDOM from "react-dom";

import Modal from "./Modal";

beforeAll(() => {
  ReactDOM.createPortal = jest.fn((element, node) => {
    return element
  })
});

it('Модалка рендерится без ошибок', () => {
  const tree = renderer
    .create(<Modal onClose={() => {
      alert('Я не понимаю, что я делаю!')
    }
    }/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})

