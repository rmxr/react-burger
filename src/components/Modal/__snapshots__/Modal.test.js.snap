// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Модалка рендерится без ошибок 1`] = `null`;
